lib/src/stream_queue.dart

```diff
@@ -120,7 +120,9 @@ class StreamQueue<T> {
   // Private generative constructor to avoid subclasses.
   StreamQueue._(this._source) {
     // Start listening immediately if we could otherwise lose events.
-    if (_source.isBroadcast) {
+    // isBroadcast can be null when using [mockito](https://pub.dev/packages/mockito)
+    // to create a mock stream.
+    if (_source?.isBroadcast == true) {
```

Comment:
The `?.` isn't necessary here. `_source` should always be non-null.

```suggestion
    if (_source.isBroadcast ?? false) {
```

That said, this feels a little sketchy to me. After NNBD the analyzer won't even want us to write it like this.

It's definitely unfortunate that this surfaces as broken tests, but I do think that the better solution is likely to make the mocks compatible rather than work around it.

@jakemac53 @lrhn - what do you think?

Reply:
Yeah I was just trying to be as safe as possible but you're right it shouldn't be necessary in this case.
I feel it would be best to be backward compatible until NNBD lands in stable then this code could be removed with the migration.